### ISSUE TITLE
common common functions ;-)

### ISSUE THREAD

**Issue by mojoaxel** [coding-horror/basic-computer-games#699]
It was decided to have a (minimal) set of common IO-functions so those don't need to be re-implemented in every game.
I'll propose to use the same function (same names) in every programming language.

That way the code get's easier to compare.

**Comment by mojoaxel** [comment]
To stay close to the original BASIC implementation I would propose the following functions (pseudo language)

* `writeLine(...messages)` - Writes content to the terminal and finishes with a linebrank.
* `async input(message)` - prints out the message followed by a space and waits for an user input.
   The promise is resolved on `enter`. The result is a string without a newline at the end.

In addition it could be useful to have the following functions:

* `write(...messages)` - Writes content to the terminal without a newline at the end.
* `tab(length)` - returns a string with the given length containing only of spaces
* `random(max, min=0)` - Returns random integer number [min...max].

**Comment by coding-horror** [comment]
yes! I think this is perfect. Just the minimum viable set of common functions, justified by at least 10 (ish) of the programs needing them..?

**Comment by ugurkupeli** [comment]
I have published a utility [crate](https://crates.io/crates/morristown) for rust users.
Currently there are only functions for printing intro/instructions and different types of user prompts - but I think that is good for our purpose.

**Comment by drewjcooper** [comment]
See the common library for .Net at `00_Common/dotnet` for my rationale and implementation of this idea. 

I like the idea of having a set of functions with names/signatures that are typical of the language's standard library, but provide the behaviour of the BASIC interpreter.

I don't like the idea of creating a DSL that looks like BASIC. Hence, I have not implemented a `Tab(int)` function in the .Net common library, or any of my game ports, preferring to stick to C#/.Net idioms.
